ui/lib/core/addon/components/search-select-with-modal.js

```diff
@@ -13,137 +12,148 @@ import { filterOptions, defaultMatcher } from 'ember-power-select/utils/group-ut
- * see the power-select docs for more information.
- *
+ * @param {boolean} [displayInherit=false] - if you need the search select component to display inherit instead of box.
  */
 export default class SearchSelectWithModal extends Component {
   @service store;
-
-  @tracked selectedOptions = null; // list of selected options
-  @tracked allOptions = null; // all possible options
-  @tracked showModal = false;
-  @tracked newModelRecord = null;
   @tracked shouldUseFallback = false;
 
-  constructor() {
-    super(...arguments);
-    this.selectedOptions = this.inputValue;
-  }
+  @tracked selectedOptions = []; // list of selected options
+  @tracked dropdownOptions = []; // options that will render in dropdown, updates as selections are added/discarded
+  @tracked showModal = false;
+  @tracked nameInput = null;
 
-  get inputValue() {
-    return this.args.inputValue || [];
+  get hidePowerSelect() {
+    return this.selectedOptions.length >= this.args.selectLimit;
   }
 
   get shouldRenderName() {
-    return this.args.shouldRenderName || false;
+    return this.args.models?.some((model) => model.includes('identity')) || this.args.shouldRenderName
+      ? true
+      : false;
   }
 
-  get excludeOptions() {
-    return this.args.excludeOptions || null;
+  addSearchText(optionsToFormat) {
+    // maps over array models from query
+    return optionsToFormat.toArray().map((option) => {
+      option.searchText = `${option.name} ${option.id}`;
+      return option;
+    });
   }
 
-  get passObject() {
-    return this.args.passObject || false;
+  formatInputAndUpdateDropdown(inputValues) {
+    // inputValues are initially an array of strings from @inputValue
+    // map over so selectedOptions are objects
+    return inputValues.map((option) => {
+      const matchingOption = this.dropdownOptions.findBy('id', option);
+      // remove any matches from dropdown list
+      this.dropdownOptions.removeObject(matchingOption);
+      return {
+        id: option,
+        name: matchingOption ? matchingOption.name : option,
+        searchText: matchingOption ? matchingOption.searchText : option,
+      };
+    });
   }
 
-  @action
-  async fetchOptions() {
-    try {
-      const queryOptions = {};
-      const options = await this.store.query(this.args.model, queryOptions);
-      this.formatOptions(options);
-    } catch (err) {
-      if (err.httpStatus === 404) {
-        if (!this.allOptions) {
-          // If the call failed but the resource has items
-          // from a different namespace, this allows the
-          // selected items to display
-          this.allOptions = [];
+  @task
+  *fetchOptions() {
+    this.dropdownOptions = []; // reset dropdown anytime we re-fetch
+    if (!this.args.models) {
+      return;
+    }
+
+    for (const modelType of this.args.models) {
+      try {
+        const queryParams = {};
+        // fetch options from the store
+        let options = yield this.store.query(modelType, queryParams);
```

Comment:
This is a total nit, but if we're only using `queryParams` for the fetch could we put it inline? there's already a lot to read here so removing an extra const helps clarify what's happening
```suggestion
        let options = yield this.store.query(modelType, {});
```